char_conv.c

```diff
@@ -23,385 +23,204 @@
 #include "mutils.h"			/* ITEMS_OF() */
 #include "gsm7_luts.h"
 
-EXPORT_DEF ssize_t convert_string (const char* in, size_t in_length, char* out, size_t out_size, char* from, char* to)
-{
-	ICONV_CONST char*	in_ptr = (ICONV_CONST char*) in;
-	size_t			in_bytesleft = in_length;
-	char*			out_ptr = out;
-	size_t			out_bytesleft = out_size - 1;
-	ICONV_T			cd = (ICONV_T) -1;
-	ssize_t			res;
-
-	cd = iconv_open (to, from);
-	if (cd == (ICONV_T) -1)
-	{
+static char lut_hex2val[] = {
+	-1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, 
+	-1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, 
+	-1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, 
+	0, 1, 2, 3, 4, 5, 6, 7, 8, 9, -1, -1, -1, -1, -1, -1, 
+	-1, 10, 11, 12, 13, 14, 15, -1, -1, -1, -1, -1, -1, -1, -1, -1, 
+	-1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, 
+	-1, 10, 11, 12, 13, 14, 15, -1, -1, -1, -1, -1, -1, -1, -1, -1, 
+	-1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, 
+	-1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, 
+	-1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, 
+	-1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, 
+	-1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, 
+	-1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, 
+	-1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, 
+	-1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, 
+	-1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1
+};
+static const char* lut_val2hex = "0123456789ABCDEF";
+
+static ssize_t convert_string(const char *in, size_t in_length, char *out, size_t out_size, char *from, char *to)
+{
+	ICONV_CONST char *in_ptr = (ICONV_CONST char*)in;
+	size_t in_bytesleft = in_length;
+	char *out_ptr = out;
+	size_t out_bytesleft = out_size - 1;
+	ICONV_T cd = (ICONV_T)-1;
+	ssize_t res;
+
+	cd = iconv_open(to, from);
+	if (cd == (ICONV_T)-1) {
 		return -2;
```

Comment:
Better explain why you're doing this.. (the `/2` in to_usc2, I guess?)

Reply:
What do you mean exactly? The -2?

The convert string function is originally from bg111 or so.